lib/chef/resource/mount.rb

```diff
@@ -42,7 +42,7 @@ class Mount < Chef::Resource
         sensitive: true
 
       property :mount_point, String, name_property: true,
-               coerce: proc { |arg| arg.chomp("/") }, # Removed "/" from the end of str, because it was causing idempotency issue.
+               coerce: proc { |arg| (arg == "/" || arg.match?(":/$"))? arg : arg.chomp("/") }, # Removed "/" from the end of str, because it was causing idempotency issue.
```

Comment:
maybe `proc { |arg| arg.count("/") > 1 && arg.chomp("/") }` ?

Reply:
I stayed with this implementation for the consistency across mount resource / provider, per https://github.com/chef/chef/blob/main/lib/chef/provider/mount.rb#L180 and https://github.com/chef/chef/blob/main/lib/chef/provider/mount/mount.rb#L206.
Do you want me to update the logic in these places too?

Reply:
okay i think after talking this out in issue triage that i'm buying the argument that the code as its written in these three places has been battle tested and does not have edge conditions that have been experienced while theoretically my suggestion could have edge conditions, so this code is fine as it is.